File: Interviewbit/Array/reverse-level-order.java
/**
 * Definition for binary tree
 * class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) {
 *      val = x;
 *      left=null;
 *      right=null;
 *     }
 * }
 */
public class Solution {
    public ArrayList<Integer> solve(TreeNode A) {
        ArrayList<Integer> ans=new ArrayList<>();
        Queue<TreeNode> q=new LinkedList<TreeNode>();
        TreeNode tmp;
        q.add(A);
        
        while(!q.isEmpty()){
            int size=q.size();
            for(int i=0;i<size;i++){
                tmp=q.peek();
                q.poll();
                if(tmp.right!=null){
                    q.add(tmp.right);
                }
                if(tmp.left!=null){
                    q.add(tmp.left);
                }
                ans.add(tmp.val);
            }
        }
        Collections.reverse(ans);
        return ans;
    }
}

File: Interviewbit/Tree/reverse-level-order.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

 vector<int> reverseOrder(TreeNode* root){
     queue<TreeNode*>q;
     TreeNode* tmp;
     vector<int>v;
     q.push(root);
     while(!q.empty()){
         v.clear();
         int size=q.size();
         for(int i=0;i<size;i++){
            tmp=q.front();
            q.pop();
            if(tmp->right) q.push(tmp->right);
            if(tmp->left) q.push(tmp->left);
            v.push_back(tmp->val);
         }
     }
     reverse(v.begin(),v.end());
     return v;
 }
 
vector<int> Solution::solve(TreeNode* A) {
    
    if(A==NULL) return {};
    return reverseOrder(A);
}
S
